pkg/datapath/linux/ipsec/ipsec_linux_test.go

```diff
@@ -226,6 +226,14 @@ func TestUpsertIPSecEndpoint(t *testing.T) {
 func TestUpsertIPSecKeyMissing(t *testing.T) {
 	setupIPSecSuitePrivileged(t)
 
+	// This should be cleaned up by previous test,
+	// but if it fails, it won't clean it up.
+	// Let's clean it up here to avoid failing this test too
+	// and causing confusion.
+	ipSecKeysGlobal[""] = nil
+	ipSecKeysGlobal["1.1.3.4"] = nil
+	ipSecKeysGlobal["1.2.3.4"] = nil
```

Comment:
Shouldn't we instead move this same cleanup code from previous tests and into `setupIPSecSuitePrivileged`'s `Cleanup` function? It seems more future proof and wouldn't depend on the order of tests.

Reply:
yea, makes sense. I cleaned it up a bit more. 

I've also added a test that checks if we create xfrmState on UpsertIPsecEndpoint. It seems to me like all tests for `UpsertIPsecEndpoint` were only checking if we report an error or not, not checking if we create a proper state.

Reply:
Thanks!

Reply:
Funny thing, seems like we were indeed not creating any xfrm state/policy because local/remote ips were the same:
https://github.com/cilium/cilium/blob/24341ba8e9f0d90af2fbfb21fa155729e532cc2f/pkg/datapath/linux/ipsec/ipsec_linux_test.go#L119-L122
https://github.com/cilium/cilium/blob/68f5f4140530ecd7af0768d459debef8ebd6fdc3/pkg/datapath/linux/ipsec/ipsec_linux.go#L932
:cry: